src/common.cpp

```diff
@@ -1873,22 +1880,18 @@ std::string get_executable_path(const char *argv0) {
     } else {
         return std::string(buff);
     }
-#else
-    // On other unixes, fall back to the Linux-ish /proc/ directory
+#elif (defined(__linux__) || defined(sun) || defined(__sun))
     ssize_t len;
     len = readlink("/proc/self/exe", buff, sizeof buff - 1);  // Linux
     if (len == -1) {
-        len = readlink("/proc/curproc/file", buff, sizeof buff - 1);  // other BSDs
-        if (len == -1) {
-            len = readlink("/proc/self/path/a.out", buff, sizeof buff - 1);  // Solaris
-        }
+        len = readlink("/proc/self/path/a.out", buff, sizeof buff - 1);  // Solaris
     }
     if (len > 0) {
         buff[len] = '\0';
         // When /proc/self/exe points to a file that was deleted (or overwritten on update!)
         // then linux adds a " (deleted)" suffix.
         // If that's not a valid path, let's remove that awkward suffix.
-        std::string buffstr{buff};
+        std::string buffstr = {buff};
```

Comment:
This was fine before. `type name {value}` is another way to initialize a variable in C++. It prevents some baffling constructor choices.

Reply:
that was actually an accident. lol